serde_arrow/src/arrow2_impl/deserialization.rs

```diff
@@ -154,7 +154,9 @@ impl BufferExtract for dyn Array {
             T::F16 => convert_primitive!(f16, F16, push_u16_cast),
             T::F32 => convert_primitive!(f32, F32, push_u32_cast),
             T::F64 => convert_primitive!(f64, F64, push_u64_cast),
+            T::Date32 => convert_primitive!(i32, Date32, push_u32_cast),
             T::Date64 => convert_primitive!(i64, Date64, push_u64_cast),
+            T::Time64(_) => convert_primitive!(i64, Time64, push_u64_cast),
```

Comment:
Is there really no unit dpependence for deserialization?

Reply:
I followed the `T::Timestamp(_, _) => convert_primitive!(i64, Date64, push_u64_cast)`, is the idea that this should return an error if the Unit is not nanoseconds or microseconds?

Reply:
Sorry. I didn't quite think it through. I think in its current state (only integers are supported), it's sufficient as is. I post a longer explantion below.